modules/date-localizer/test/spec.coffee

```diff
@@ -3,79 +3,221 @@ describe 'dateTimeLocalizerMoment', ->
 
 describe 'dateTimeLocalizer', ->
   localizer = hx.dateTimeLocalizer()
-
+  zeroPad = hx.format.zeroPad(2)
   testDate = new Date(1452130200000) # Thu Jan 07 2016 01:30:00 GMT+0000 (GMT)
   zeroHourDate = new Date(1452124800000) # Thu Jan 07 2016 00:00:00 GMT+0000 (GMT)
   invalidDate = new Date('Invalid Date')
+  # 0 - London, -120 - Sofia, -60 Krakow, -540 Tokio, 420 Pinal (Arizona/United states)
+  timezonesOfsets = [0, -120, -60, -540, 420]
+
+  getHexaGonTimeZoneOffset = ->
+    hexagonTimeZoneOffset = hx.preferences.timezone().replace(/^UTC\+?/,'').split(":").map (item, index) ->
+      if index is 0
+        Number(item) * 60
+      else
+        Number(item)
+    .reduce (current, previous) ->
+      current + previous
+    hexagonTimeZoneOffset
+
+  # This is helper function. It accepts
+  # function as a argument and execute it
+  # in all timezones that is provided
+  executeFunctionInAllTimeZones = (func) ->
+    for currentOffset in timezonesOfsets
+      oldDate = Date
+      # This replace of date constructor
+      # working properly under browser
+      # but under test this don't have
+      # effect. I tried big set of things
+      # with sinon also.
+      # The atempt below with
+      # hx.preferences.timezone(timeZone)
+      # really work.
+      Date = do (newDate = Date) ->
+        ->
+          dateResult = new (Function.prototype.bind.apply(newDate, arguments))
+          dateResult.setTime(dateResult.getTime() + (currentOffset * 60000))
+          dateResult
+      func()
+      Date = oldDate
```

Comment:
Actually code from 37nd to 43nd line has no effect under nodejs. In the browser it works perfectly. There is no errors and etc but Date object behaviour remains unchanged. I tried many things like
```
sinon.useFakeTimers calculatedMiliseconds, 'Date'
```
but this also not helped. I left the code here to show the issue. We can remove it. The code bellow - from 44 to 48
```
     oldTimeZone = hx.preferences.timezone()
     timeZone = hx._.preferences.defaultTimezoneLookup(currentOffset)
     hx.preferences.timezone(timeZone)
     func()
     hx.preferences.timezone(oldTimeZone)
```
line working properly but in fact this changes Hexagon internal time-zone not browser time-zone.